ophyd/areadetector/cam.py

```diff
@@ -1289,6 +1290,171 @@ class RoperDetectorCam(CamBase):
     roper_shutter_mode = ADCpt(SignalWithRBV, "RoperShutterMode")
 
 
+class TimePix3Chip(Device):
+
+    def __init__(self, chip_number):
+        self.chip_number = chip_number
+        self.cp_pll = ADCpt(EpicsSignalRO, f'{chip_number}_CP_PLL_RBV')
+        self.s1_off = ADCpt(EpicsSignalRO, f'{chip_number}_S1_OFF_RBV')
+        self.s1_on = ADCpt(EpicsSignalRO, f'{chip_number}_S1_ON_RBV')
+        self.s2_off = ADCpt(EpicsSignalRO, f'{chip_number}_S2_OFF_RBV')
+        self.s2_on = ADCpt(EpicsSignalRO, f'{chip_number}_S2_ON_RBV')
+        self.ikrum = ADCpt(EpicsSignalRO, f'{chip_number}_Ikrum_RBV')
+        self.ixel_dac = ADCpt(EpicsSignalRO, f'{chip_number}_PixelDAC_RBV')
+        self.preamp_off = ADCpt(EpicsSignalRO, f'{chip_number}_Preamp_OFF_RBV')
+        self.preamp_on = ADCpt(EpicsSignalRO, f'{chip_number}_Preamp_ON_RBV')
+        self.t_pbuffer_in = ADCpt(EpicsSignalRO, f'{chip_number}_TPbufferIn_RBV')
+        self.t_pbuffer_out = ADCpt(EpicsSignalRO, f'{chip_number}_TPbufferOut_RBV')
+        self.pll_vcntrl = ADCpt(EpicsSignalRO, f'{chip_number}_PLL_Vcntrl_RBV')
+        self.v_preamp_ncas = ADCpt(EpicsSignalRO, f'{chip_number}_VPreamp_NCAS_RBV')
+        self.vtp_coarse = ADCpt(EpicsSignalRO, f'{chip_number}_VTP_coarse_RBV')
+        self.vtp_fine = ADCpt(EpicsSignalRO, f'{chip_number}_VTP_fine_RBV')
+        self.vfbk = ADCpt(EpicsSignalRO, f'{chip_number}_Vfbk_RBV')
+        self.vth_coarse = ADCpt(EpicsSignalRO, f'{chip_number}_Vth_coarse_RBV')
+        self.vth_fine = ADCpt(EpicsSignalRO, f'{chip_number}_Vth_fine_RBV')
+        self.adjust = ADCpt(EpicsSignalRO, f'{chip_number}_Adjust_RBV')
+        self.layout = ADCpt(EpicsSignalRO, f'{chip_number}_Layout_RBV')
+
+    
+class TimePix3DetectorCam(CamBase):
+    _html_docs = ['TimePix3Doc.html']
+    _default_configuration_attrs = (
+        CamBase._default_configuration_attrs
+    )
+
+    chip0 = TimePix3Chip(0)
+    chip1 = TimePix3Chip(1)
+    chip2 = TimePix3Chip(2)
+    chip3 = TimePix3Chip(3)
```

Comment:
```suggestion
    chip0 = Cpt(TimePix3Chip, 0)
    chip1 = Cpt(TimePix3Chip, 1)
    chip2 = Cpt(TimePix3Chip, 2)
    chip3 = Cpt(TimePix3Chip, 3)
```

Reply:
These need to be components or the `TimePix3Chip` instances will be instantiated on the class instance and shared across all instances of the device (and also miss getting the PV information).